src/Microsoft.AspNetCore.Mvc.Razor.Extensions/ModelDirective.cs

```diff
@@ -14,7 +14,11 @@ public static class ModelDirective
         public static readonly DirectiveDescriptor Directive = DirectiveDescriptor.CreateDirective(
             "model",
             DirectiveKind.SingleLine,
-            builder => builder.AddTypeToken());
+            builder =>
+            {
+                builder.AddTypeToken();
+                builder.Usage = DirectiveUsage.SinglePreContent;
```

Comment:
Thought about that approach but didn't feel like it deserved its own argument since having an unrestricted directive was consistent with what we've always had in the past. Aka, users previous expectations of directives is the default experience.